test/util/TestWrapper.js

```diff
@@ -126,58 +126,61 @@ class TestWrapper {
-                    placeholder = placeholder.replace(/[{}]/g, '');
-                    if (Array.isArray(data[placeholder])) {
-                        forEach(data[placeholder], (contentData) => {
-                            const contentValueReplace = contentValue.replace(placeholderRegex, contentData);
-                            contentToSubstitute.push(contentValueReplace);
-                        });
-                        contentToSubstitute[key] = 'undefined';
-                    } else {
-                        contentValue = contentValue.replace(placeholderRegex, data[placeholder]);
-                        contentToSubstitute[key] = contentValue;
-                    }
-                });
-            } else {
-                contentToSubstitute[key] = contentValue;
-            }
-        });
+        Object.entries(contentToSubstitute)
+            .forEach(([key, contentValue]) => {
+                contentValue = contentValue.replace(/\n/g, '<br />\n');
+                const contentValueMatch = contentValue.match(/{(.*?)}/g);
+                if (contentValueMatch) {
+                    contentValueMatch.forEach(placeholder => {
+                        if (Array.isArray(data[placeholder])) {
+                            data[placeholder].forEach(contentData => {
+                                const contentValueReplace = contentValue.replace(placeholder, contentData);
+                                contentToSubstitute.push(contentValueReplace);
+                            });
+                            contentToSubstitute[key] = 'undefined';
+                        } else {
+                            contentValue = contentValue.replace(placeholder, data[placeholder]);
+                            contentToSubstitute[key] = contentValue;
+                        }
+                    });
+                } else {
+                    contentToSubstitute[key] = contentValue;
+                }
+            });
         return contentToSubstitute.filter(content => content !== 'undefined');
     }
 
     substituteErrorsContent(data, contentToSubstitute, type) {
-        forEach(Object.entries(contentToSubstitute), ([contentKey, contentValue]) => {
-            forEach(Object.entries(contentValue[type]), ([errorMessageKey, errorMessageValue]) => {
-                forEach(errorMessageValue.match(/\{(.*?)\}/g), (placeholder) => {
-                    const placeholderRegex = new RegExp(placeholder, 'g');
-                    contentToSubstitute[contentKey][type][errorMessageKey] = contentToSubstitute[contentKey][type][errorMessageKey].replace(placeholderRegex, data[placeholder]);
-                });
+        Object.entries(contentToSubstitute).forEach(([contentKey, contentValue]) => {
+            Object.entries(contentValue[type]).forEach(([errorMessageKey, errorMessageValue]) => {
+                const errorMessageValueMatch = errorMessageValue.match(/{(.*?)}/g);
```

Comment:
Just to confirm - there will only every be a single errorMessageValue for contentValue[type]
As previous code reads that there could be multiple error messages for contentValue

Reply:
Not sure what you're saying here, the substance of the code hasn't changed here :-/